python/tests/test.py

```diff
@@ -615,24 +615,30 @@ def test_basic(self):
 
         self.assertEqual(self.config.input('ex_linear').input_type.name, 'current_clamp')
         self.assertEqual(self.config.input('ex_linear').module.name, 'linear')
-        self.assertEqual(self.config.input('ex_linear').amp_start, 0.15)
-        self.assertEqual(self.config.input('ex_linear').amp_end, 0.15)
         self.assertEqual(self.config.input('ex_linear').delay, 0)
         self.assertEqual(self.config.input('ex_linear').duration, 15)
         self.assertEqual(self.config.input('ex_linear').node_set, "Column")
-        self.assertEqual(self.config.input('ex_rel_linear').percent_start, 80)
-        self.assertEqual(self.config.input('ex_rel_linear').percent_end, 20)
-        self.assertEqual(self.config.input('ex_pulse').width, 1)
-        self.assertEqual(self.config.input('ex_pulse').frequency, 80)
-        self.assertEqual(self.config.input('ex_noise_meanpercent').mean_percent, 0.01)
-        self.assertEqual(self.config.input('ex_noise_meanpercent').mean, None)
-        self.assertEqual(self.config.input('ex_noise_mean').mean, 0)
-        self.assertEqual(self.config.input('ex_noise_mean').mean_percent, None)
-        self.assertEqual(self.config.input('ex_rel_shotnoise').random_seed, self.config.run.random_seed)
-        self.assertEqual(self.config.input('ex_rel_shotnoise').dt, 0.25)
-        self.assertEqual(self.config.input('ex_replay').spike_file,
+        self.assertEqual(self.config.input('ex_linear').parameters.amp_start, 0.15)
```

Comment:
This PR is a nice improvement.  I have just one concern:
In neurodamus, currently we retrieve all the parsed values from libsonata via function like
`getattr(libsonata_obj, sonata_key)`
This PR changes the API a bit. For certain parameters of an input object, I need to adapt neurodamus to call `getattr(libsonata_obj.parameters, amp_start)`. Is it possible to avoid the layer `parameters`?
cc @ferdonline

Reply:
Yeah, that could be done.  Right now it's clear what is common to all stimulus (`module/inputType/delay/duration/nodeSet`) and what is unique to each.

> getattr(libsonata_obj.parameters, amp_start)

Is there a reason why this can't be `libsonata_obj.parameters.amp_start`?

Reply:
>Is there a reason why this can't be libsonata_obj.parameters.amp_start?

It is to avoid additional coding in neurodamus to adapt this change. It would be great if we can call `libsonata_obj.<key_name>` for both common and unique keys. Please let me know if it is feasible. Thanks!

Reply:
It's feasible, from an ergonomic point of view it's less clean, especially on the C++ side, I would say - one has to either use `visit` because one can't look at the value of the `module` before picking the correct variant.

I'll do and push the change, and we can decide then.

Reply:
> I'll do and push the change, and we can decide then.

I did the change, let me know what you think.

Reply:
Yes, it looks like what I was asking for. I will review it today.